packages/config/src/Config.ts

```diff
@@ -488,8 +494,35 @@ function ensureConfigHasDefaultValues({
     platforms = getSupportedPlatforms(projectRoot);
   }
 
+  const defaultIos = expWithDefaults.ios
+    ? ({
+        ...expWithDefaults.ios,
+        runtimeVersion: getRuntimeVersion(expWithDefaults, 'ios'),
+        buildNumber: getBuildNumber(expWithDefaults),
+      } as any) //TODO(JJ) remove this cast in SDK 43 https://linear.app/expo/issue/ENG-1869/remove-tempruntimeversion-in-expoconfig
+    : undefined;
+  const defaultAndroid = expWithDefaults.android
+    ? ({
+        ...expWithDefaults.android,
+        runtimeVersion: getRuntimeVersion(expWithDefaults, 'android'),
+        versionCode: getVersionCode(expWithDefaults),
+      } as any) //TODO(JJ) remove this cast in SDK 43 https://linear.app/expo/issue/ENG-1869/remove-tempruntimeversion-in-expoconfig
+    : undefined;
+  // TODO(JJ) remove top level runtimeVersion in favor of platform specific definitions
+  // While there is code that expects either a string or undefined here, pass only non-object runtime Version
+  const topLevelRuntimeVersion =
+    typeof expWithDefaults.runtimeVersion === 'string' ? expWithDefaults.runtimeVersion : undefined;
+
   return {
-    exp: { ...expWithDefaults, sdkVersion, platforms },
+    exp: {
+      ...expWithDefaults,
+      version: getVersion(expWithDefaults),
```

Comment:
@EvanBacon & @esamelson does it feel right to you guys to be falling back to the default `version`, `versionCode`, and `buildNumber` fields here?

Since we are formatting the `runtimeVersion` which is a function of these values, it seems correct to ensure that these values are explicit.